x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/query/ShapeQueryBuilder.java

```diff
@@ -100,23 +100,19 @@ protected ShapeQueryBuilder newShapeQueryBuilder(String fieldName, Supplier<Geom
         return new ShapeQueryBuilder(fieldName, shapeSupplier, indexedShapeId);
     }
 
-    @Override
-    public String queryFieldType() {
-        return ShapeFieldMapper.CONTENT_TYPE;
-    }
-
     @Override
     @SuppressWarnings({ "rawtypes" })
-    protected List validContentTypes() {
+    protected List<String> validContentTypes(){
         return Arrays.asList(ShapeFieldMapper.CONTENT_TYPE);
     }
 
     @Override
     @SuppressWarnings({ "rawtypes" })
     public Query buildShapeQuery(QueryShardContext context, MappedFieldType fieldType) {
-        if (fieldType.typeName().equals(ShapeFieldMapper.CONTENT_TYPE) == false) {
+        if (validContentTypes().contains(fieldType.typeName()) == false) {
```

Comment:
```suggestion
        List<String> validContentTypes = validContentTypes()
        if (validContentTypes.contains(fieldType.typeName()) == false) {
```

Reply:
I was afraid of this danger biting us. I'm not much a fan of `validContentTypes()` returning a new `List` instance for every call. So maybe we should consider making a static variable within each of the derived builders (that only calls this method once)? I'm okay changing that in a follow up PR, but lets change this method now to only call `validContentTypes` once.